examples/camera/main.go

```diff
@@ -0,0 +1,262 @@
+)
+
+func init() {
+	// Decode image from a byte slice instead of a file so that
+	// this example works in any working directory.
+	// If you want to use a file, there are some options:
+	// 1) Use os.Open and pass the file to the image decoder.
+	//    This is a very regular way, but doesn't work on browsers.
+	// 2) Use ebitenutil.OpenFile and pass the file to the image decoder.
+	//    This works even on browsers.
+	// 3) Use ebitenutil.NewImageFromFile to create an ebiten.Image directly from a file.
+	//    This also works on browsers.
+	img, _, err := image.Decode(bytes.NewReader(images.Tiles_png))
+	if err != nil {
+		log.Fatal(err)
+	}
+	tilesImage, _ = ebiten.NewImageFromImage(img, ebiten.FilterDefault)
+}
+
+type Camera struct {
+	ViewPort    f64.Vec2
+	Position    f64.Vec2
+	Zoom        f64.Vec2
+	Rotation    float64
+	worldMatrix ebiten.GeoM
+}
+
+func (c *Camera) String() string {
+	return fmt.Sprintf(
+		"T: %.1f, R: %.1f, S: %.1f",
+		c.Position, c.Rotation, c.Zoom,
+	)
+}
+
+func (c *Camera) viewportCenter() f64.Vec2 {
+	return f64.Vec2{
+		c.ViewPort[0] * 0.5,
+		c.ViewPort[1] * 0.5,
+	}
+}
+
+func (c *Camera) updateMatrix() {
+	c.worldMatrix.Reset()
+	c.worldMatrix.Translate(-c.Position[0], -c.Position[1])
+	// We want to scale and rotate around center of image / screen
+	c.worldMatrix.Translate(-c.viewportCenter()[0], -c.viewportCenter()[1])
+	c.worldMatrix.Scale(c.Zoom[0], c.Zoom[1])
+	c.worldMatrix.Rotate(c.Rotation)
+	c.worldMatrix.Translate(c.viewportCenter()[0], c.viewportCenter()[1])
+}
+
+func (c *Camera) Render(world, screen *ebiten.Image) error {
+	c.updateMatrix()
+	return screen.DrawImage(world, &ebiten.DrawImageOptions{
+		GeoM: c.worldMatrix,
+	})
+}
+
+func (c *Camera) ScreenToWorld(posX, posY int) (float64, float64) {
+	c.updateMatrix()
+	inverseMatrix := c.worldMatrix
+	if inverseMatrix.IsInvertible() {
+		inverseMatrix.Invert()
+		return inverseMatrix.Apply(float64(posX), float64(posY))
+	} else {
+		// When scaling it can happend that matrix is not invertable
+		return math.NaN(), math.NaN()
+	}
+}
+
+func (c *Camera) Reset() {
+	c.Position[0] = 0
+	c.Position[1] = 0
+	c.Rotation = 0
+	c.Zoom[0] = 1
+	c.Zoom[1] = 1
+
+	c.updateMatrix()
+}
+
+type Game struct {
+	layers [][]int
+	world  *ebiten.Image
+	camera Camera
+}
+
+func (g *Game) Update(screen *ebiten.Image) error {
+	if ebiten.IsKeyPressed(ebiten.KeyA) || ebiten.IsKeyPressed(ebiten.KeyLeft) {
+		g.camera.Position[0] -= 1
+	}
+	if ebiten.IsKeyPressed(ebiten.KeyD) || ebiten.IsKeyPressed(ebiten.KeyRight) {
+		g.camera.Position[0] += 1
+	}
+	if ebiten.IsKeyPressed(ebiten.KeyW) || ebiten.IsKeyPressed(ebiten.KeyUp) {
+		g.camera.Position[1] -= 1
+	}
+	if ebiten.IsKeyPressed(ebiten.KeyS) || ebiten.IsKeyPressed(ebiten.KeyDown) {
+		g.camera.Position[1] += 1
+	}
+
+	if ebiten.IsKeyPressed(ebiten.KeyQ) {
+		g.camera.Zoom[0] -= .1
+		g.camera.Zoom[1] -= .1
+	}
+	if ebiten.IsKeyPressed(ebiten.KeyE) {
+		g.camera.Zoom[0] += .1
+		g.camera.Zoom[1] += .1
```

Comment:
1) Adding / reducing 0.1 seems aggressive. What about 0.01?
2) Adding / reducing values from float values can cause numerical errors. What about keeping the zoom factor as an integer and assign a calculated value?
3) (optional) Rather than a linear zoom change, I'd adopt an exponential way:

`g.camera.Zoom[0] = math.Pow(1.01, g.zoomFactor)`